tests/template/cockroachdb_helm_template_test.go

```diff
@@ -0,0 +1,283 @@
+package template
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+)
+
+// TestHelmSelfCertSignerTemplateRenderedSuccessfully contains the successful rendered objects by the cockroachdb helm
+// chart with selfSigner enabled.
+func TestHelmSelfCertSignerTemplateRenderedSuccessfully(t *testing.T) {
```

Comment:
Can we have some scenarios where we check more than the namespace and check that parts of the template guarded by `if/then` statements are what we expect?

Reply:
I have added the if else check in the `statefulset.yaml` and `job.init.yaml`. These two files changes based on selfSigner.enabled value. 
Also, added tests for disable the selfSigner and check the self signer resources are not templated.
Implemented table driven tests to optimize the test code.